### PR TITLE
HW

### PR DESCRIPTION
**Pull request by Sp0ozy**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package main
+
+type node struct {
+	val         int
+	left, right *node
+}
+
+func levelOrderTraversal(root *node, sink func(v int)) (res []int) {
+	level := make(map[int][]int)
+	var visit func(root *node, level int, levelMapNode *map[int][]int)
```
Comment (zayac) [inline main.go:10]: Maps are passed by a reference to a function, so a pointer to the map is not needed.

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
+package main
```
Comment (zayac) [inline main_test.go:1]: There are many corner cases for both programs, so I expect more tests.
